src/components/TheCanvas.vue

```diff
@@ -63,69 +68,47 @@ export default class TheCanvas extends Vue {
           }
         }
       }, true)
+
       this.socket.onmessage = (data: MessageEvent) => {
         try {
-          const receivedData = JSON.parse(data.data).payload
-          if (receivedData.type && receivedData.data.e) {
-            this.enableTool(receivedData.type)
-            this.throttleToolRender(receivedData.data.e, true)
+          const canvasElement: CanvasElement = JSON.parse(data.data).payload
+          if (canvasElement.immediate) {
+            // render it on the canvas
+            const foundTool = this.tools.find((tool: Tool) => tool.name === canvasElement.tool.name)
+            if (foundTool && foundTool.renderCanvas && canvasElement.jti !== 'SAM') {
```

Comment:
Uh, hardcoding detected.

Reply:
because we haven't got the JTI set up yet! So I put is some hardcode until you do that work ;)